dlt/common/libs/pyarrow.py

```diff
@@ -173,3 +173,29 @@ def get_row_count(parquet_file: TFileOrPath) -> int:
 
 def is_arrow_item(item: Any) -> bool:
     return isinstance(item, (pyarrow.Table, pyarrow.RecordBatch))
+
+
+TNewColumns = Sequence[Tuple[pyarrow.Field, Callable[[pyarrow.RecordBatch], Iterable[Any]]]]
+
+def pq_stream_with_new_columns(
+    parquet_file: TFileOrPath, columns: TNewColumns, batch_size: int = 100
+) -> Iterator[pyarrow.RecordBatch]:
+    """Add a column to the parquet file.
+
+    Args:
+        parquet_file: path or file object to parquet file
+        columns: list of columns to add in the form of (`pyarrow.Field`, column_value_callback)
+            The callback should accept a `pyarrow.RecordBatch` and return an iterable of values for the column.
+        batch_size: batch size to use when reading the parquet file. Defaults to 100.
+
+    Yields:
+        `pyarrow.RecordBatch` objects with the new columns added.
+    """
+    with pyarrow.parquet.ParquetFile(parquet_file) as reader:
```

Comment:
I just run a few tests and rewriting those parquets is slower than rewriting them as JSON. probably could be expected...